### PR TITLE
Rename tenant to workload

### PR DESCRIPTION
**Pull request by tfussell**
We recently decided to follow Cluster API conventions and call our tenant clusters workload clusters and our control planes management clusters. This isn't strictly necessary in our code, but I hope it will reduce mental burden when making future changes and improve consistency.

I also removed two unused flags that I noticed.

### CODE REVIEW

File: helm/kvm-operator/templates/configmap.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ data:
         ntp:
           servers: {{ .Values.Installation.V1.Provider.KVM.NTP.Servers }}
         {{- if .Values.Installation.V1.Guest }}
-        tenant:
+        workload:
```
Comment (tfussell) [inline helm/kvm-operator/templates/configmap.yaml:36]: This is the only change I could conceivably see having an impact. As far as I can tell, we don't override this in `installations` for any customer though.

**Review by yulianedyalkova — APPROVED** [review]

File: flag/service/workload/tenant.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package workload
```
Comment (stone-z) [inline flag/service/workload/tenant.go:1]: Maybe this file should also be named `workload.go`

**Review by MarcelMue — APPROVED** [review]
Seems fine. The change in flags should be tested.

**Review by stone-z — APPROVED** [review]
Sounds good

File: flag/service/workload/tenant.go (hunk 2 above)
Comment (tfussell) [reply flag/service/workload/tenant.go]: Thanks, I searched in files but didn't look at filenames. This is fixed now.

**Comment by MarcelMue** [comment]
Is there a blocker to merge this?

**Comment by tfussell** [comment]
Tested running on `gorgoth`. No issues with flag renaming. Finally merging.

**Pull request merged**